### PR TITLE
[Infra] Increase test parallelism

### PR DESCRIPTION
**Pull request by nastra**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (Infra)

## Description

Increase test JVM parallelization to 4.

## How was this patch tested?

CI tests.

## Does this PR introduce _any_ user-facing changes?

No.

### CODE REVIEW

File: project/TestParallelization.scala [hunk 1]
```diff
@@ -48,7 +48,8 @@ object TestParallelization {
    */
   lazy val simpleGroupingStrategySettings = Seq(
     Test / forkTestJVMCount := {
-      sys.env.get("TEST_PARALLELISM_COUNT").map(_.toInt).getOrElse(4)
+      sys.env.get("TEST_PARALLELISM_COUNT").map(_.toInt)
```
Comment (tdas) [inline project/TestParallelization.scala:51]: we dont need to change this for speeding up github tests.. isnt it?

Comment (tdas) [reply project/TestParallelization.scala:51]: if not, lets separate out user-facing defaults from github tests.
each carry a different risk. increasing parallelism too much can lead to flakiness, and so using all available processors may be problematic.

**Comment by tdas** [comment]
how much benefit are you getting?

**Comment by nastra** [comment]
> how much benefit are you getting?

spark tests with Scala 2.13 are down to 3h and the same tests with Scala 2.12 are down to 3:30h

File: project/TestParallelization.scala (hunk 1 above)
Comment (nastra) [reply project/TestParallelization.scala]: makes sense, I've reverted this for now

**Comment by tdas** [comment]
With the sharing bringing it down already to 2 hours, how much additional
benefits does this pr provide?

**Comment by scottsand-db** [comment]
Before this PR:
- delta-spark against Spark 3.5.2 -- Scala 2.12 :: 94, 124, 117 -- AVG: 111.67
- delta-spark against Spark 3.5.2 -- Scala 2.13 :: 72, 108, 93 -- AVG 91.00

With this PR:
- delta-spark against Spark 3.5.2 -- Scala 2.12 :: 84, 115, 95 -- AVG: 98.00
- delta-spark against Spark 3.5.2 -- Scala 2.13 ::  65, 93, 78 -- AVG: 78.67

**Pull request merged**